register.go

```diff
@@ -17,55 +17,148 @@ func getConsumerUUID() (string, error) {
+		return "", unpackErrorMsg(err)
 	}
 	return uuid, nil
 }
 
-func registerPassword(username, password, organization, serverURL string) error {
+// Organization is structure containing information about RHSM organization (sometimes called owner)
+// JSON document returned from candlepin server can have the following format. We care only about key,
+// but it can be extended and more information can be added to the structure in the future.
+//{
+//    "created": "2022-11-02T16:00:23+0000",
+//    "updated": "2022-11-02T16:00:48+0000",
+//    "id": "4028face84391264018439127db10004",
+//    "displayName": "Donald Duck",
+//    "key": "donaldduck",
+//    "contentPrefix": null,
+//    "defaultServiceLevel": null,
+//    "logLevel": null,
+//    "contentAccessMode": "org_environment",
+//    "contentAccessModeList": "entitlement,org_environment",
+//    "autobindHypervisorDisabled": false,
+//    "autobindDisabled": false,
+//    "lastRefreshed": "2022-11-02T16:00:48+0000",
+//    "parentOwner": null,
+//    "upstreamConsumer": null
+//}
+type Organization struct {
+	Key string `json:"key"`
+}
+
+// unpackOrgs tries to unpack list organization from JSON document returned by D-Bus method GetOrgs.
+// When it is possible to unmarshal the JSON document, then return list of organization keys (IDs).
+// When it is not possible to get list of organizations, then return empty slice and error.
+func unpackOrgs(s string) ([]string, error) {
+	var orgs []string
+
+	var organizations []Organization
+
+	err := json.Unmarshal([]byte(s), &organizations)
+	if err != nil {
+		return orgs, err
+	}
+
+	for _, org := range organizations {
+		orgs = append(orgs, org.Key)
+	}
+
+	return orgs, nil
+}
+
+// registerUsernamePassword tries to register system against candlepin server (Red Hat Management Service)
+// username and password are mandatory. When organization is not obtained, then this method
+// returns list of available organization and user can select one organization from the list.
+func registerUsernamePassword(username, password, organization, serverURL string) ([]string, error) {
+	var orgs []string
 	if serverURL != "" {
 		if err := configureRHSM(serverURL); err != nil {
-			return fmt.Errorf("cannot configure RHSM: %w", err)
+			return orgs, fmt.Errorf("cannot configure RHSM: %w", err)
 		}
 	}
 
 	conn, err := dbus.SystemBus()
 	if err != nil {
-		return err
+		return orgs, err
 	}
 
 	uuid, err := getConsumerUUID()
 	if err != nil {
-		return err
+		return orgs, err
 	}
 	if uuid != "" {
-		return fmt.Errorf("warning: the system is already registered")
+		return orgs, fmt.Errorf("warning: the system is already registered")
 	}
 
 	registerServer := conn.Object("com.redhat.RHSM1", "/com/redhat/RHSM1/RegisterServer")
 
 	var privateDbusSocketURI string
-	if err := registerServer.Call("com.redhat.RHSM1.RegisterServer.Start", dbus.Flags(0), "").Store(&privateDbusSocketURI); err != nil {
-		return err
+	if err := registerServer.Call(
```

Comment:
Are you making these formatting changes yourself? Or are you using an IDE that's formatting these long function call expressions automatically into new-lines-by-parameter? I don't believe that `go fmt` cares one way or another, but including them in a patch like this just adds unnecessary change to the patch set.

Reply:
Yes, I'm making these changes myself. I consider too long lines as hard to read. Should I try to split it to another PR/commit? Or is mentioning this change in commit message enough?

Reply:
I'm fine moving to a function call style like this. Could you move them into a separate commit? The same PR is fine, but they are a separate change from the logic changes adding the organization checking code. Does `go fmt` change the number of tabs/spaces used when you format the function calls like this?

Reply:
`go fmt` is the de facto standard, so we should use it. I'm surprised Goland does not run `go fmt` before saving. One of the Go proverbs:

> Gofmt's style is no one's favorite, yet gofmt is everyone's favorite.

https://www.youtube.com/watch?v=PAAkCSZUG1c&t=8m43s

Reply:
I fixed the indentation using `go fmt`. I tried to split commit into two commits, but it would require re-implementing two steps of changes again, because there were two changes tangled together. I really don't like being slave of nice git history... I tried to describe all changes in commit message.